cmd/krel/cmd/release_notes.go

```diff
@@ -239,67 +365,204 @@ func createDraftPR(tag string, result *releaseNotesResult) error {
+	}
+
+	if err := pushToRemote(sigReleaseRepo, branchname); err != nil {
+		return errors.Wrapf(err, "pushing repo to %v", userForkName)
+	}
+
+	// TODO: Call github API and create PR against k/sig-release
+	return nil
+}
+
+// prepareFork Prepare a branch a repo
+func prepareFork(branchName, repoPath, upstreamOrg, upstreamRepo, myOrg, myRepo string) (repo *git.Repo, err error) {
+	// checkout the upstream repository
+	logrus.Infof("cloning/updating repository %s/%s", upstreamOrg, upstreamRepo)
+
+	repo, err = git.CloneOrOpenGitHubRepo(
+		repoPath, upstreamOrg, upstreamRepo, true,
+	)
 	if err != nil {
-		return errors.Wrap(err, "cloning k/sig-release")
+		return nil, errors.Wrapf(err, "cloning %s/%s", upstreamOrg, upstreamRepo)
 	}
 
 	// test if the fork remote is already existing
-	const remote = "userfork"
 	url, err := git.GetRepoURL(
-		releaseNotesOpts.draftOrg, releaseNotesOpts.draftRepo, true,
+		myOrg, myRepo, true,
 	)
 	if err != nil {
-		return errors.Wrap(err, "unable to get repository URL")
+		return nil, errors.Wrap(err, "unable to get repository URL")
 	}
-	if sigReleaseRepo.HasRemote(remote, url) {
+	if repo.HasRemote(userForkName, url) {
 		logrus.Infof(
 			"Using already existing remote %v (%v) in repository",
-			remote, url,
+			userForkName, url,
 		)
 	} else {
 		// add the user's fork as a remote
-		err = sigReleaseRepo.AddRemote(remote, releaseNotesOpts.draftOrg, releaseNotesOpts.draftRepo)
+		err = repo.AddRemote(userForkName, myOrg, myRepo)
 		if err != nil {
-			return errors.Wrap(err, "adding users fork as remote repository")
+			return nil, errors.Wrap(err, "adding user's fork as remote repository")
 		}
 	}
 
 	// verify the branch doesn't already exist on the user's fork
-	err = sigReleaseRepo.HasRemoteBranch(branchname)
+	err = repo.HasRemoteBranch(branchName)
 	if err == nil {
-		return errors.Errorf("remote repo already has a branch named %s", branchname)
+		return nil, errors.Errorf("remote repo already has a branch named %s", branchName)
 	}
 
 	// checkout the new branch
-	err = sigReleaseRepo.Checkout("-b", branchname)
+	err = repo.Checkout("-b", branchName)
 	if err != nil {
-		return errors.Wrapf(err, "creating new branch %s", branchname)
+		return nil, errors.Wrapf(err, "creating new branch %s", branchName)
 	}
 
-	// generate the notes
-	targetdir := filepath.Join(sigReleaseRepo.Dir(), "releases", fmt.Sprintf("release-%d.%d", s.Major, s.Minor))
-	logrus.Debugf("Release notes markdown will be written to %s", targetdir)
-	err = ioutil.WriteFile(filepath.Join(targetdir, draftFilename), []byte(result.markdown), 0644)
+	return repo, nil
+}
+
+// addReferenceToAssetsFile adds a new entry in the assets.ts file in repoPath to include newJsonFile
+func addReferenceToAssetsFile(repoPath, newJSONFile string) error {
+	file, err := os.Open(filepath.Join(
+		repoPath, "src", "environments", "assets.ts",
+	))
```

Comment:
I would make `filepath.Join(repoPath, "src", "environments", "assets.ts")` a new variable and reuse it in line 461